test/fuzz/random.h

```diff
@@ -0,0 +1,101 @@
+#include <memory>
+#include <random>
+#include <utility>
+#include <vector>
+
+#include "envoy/common/random_generator.h"
+
+#include "common/common/assert.h"
+#include "common/common/logger.h"
+
+namespace Envoy {
+namespace Random {
+
+class PsuedoRandomGenerator64 : public RandomGenerator {
+public:
+  PsuedoRandomGenerator64() = default;
+  ~PsuedoRandomGenerator64() override = default;
+
+  void initializeSeed(uint64_t seed) { prng_ = std::make_unique<std::mt19937_64>(seed); }
+
+  // RandomGenerator
+  uint64_t random() override {
+    // Makes sure initializeSeed() was already called
+    ASSERT(prng_ != nullptr);
+    const uint64_t to_return = (*prng_)();
+    ENVOY_LOG_MISC(trace, "random() returned: {}", to_return);
+    return to_return;
+  }
+  std::string uuid() override { return ""; }
+  std::unique_ptr<std::mt19937_64> prng_;
+};
+
+} // namespace Random
+
+namespace Fuzz {
+class ProperSubsetSelector {
+public:
+  ProperSubsetSelector(const std::string& random_bytestring)
+      : random_bytestring_(random_bytestring) {}
+
+  /**
+   * This function does proper subset selection on a certain number of elements. It returns a vector
+   * of vectors of bytes. Each vector of bytes represents the indexes of a single subset. The
+   * "randomness" of the subset that the class will use is determined by a bytestring passed into
+   * the class. Example: call into function with a vector {3, 5} representing subset sizes, and 15
+   * as number_of_elements. This function would return something such as {{3, 14, 7}, {2, 1, 13, 8,
+   * 6}}
+   */
+
+  std::vector<std::vector<uint8_t>>
+  constructSubsets(const std::vector<uint32_t>& number_of_elements_in_each_subset,
```

Comment:
tacking on a comment nit before submission:
can you document if there are any requirements on the sizes of the params?
(for eg, does number_of_elements have to be greater than sum(number_of_elements_in_each_subset)? Looking through the usage in the fuzzer (hosts made can be less than the numbers in each locality) and the code it looks like it doesn't have to be, but then you don't guarantee that the returned subsets actually have the size. worth documenting).

Reply:
Yeah, good point, it's "smooth" and it doesn't guarantee returned size.